modules/renter/repair.go

```diff
@@ -182,6 +182,52 @@ func (f *file) expiringContracts(height types.BlockHeight) []fileContract {
 	return expiring
 }
 
+// offlineChunks returns the chunks belonging to "offline" hosts -- hosts that
+// do not meet uptime requirements. Importantly, only chunks missing more than
+// half their redundancy are returned.
+func (f *file) offlineChunks(hdb hostDB) map[uint64][]uint64 {
+	f.mu.RLock()
+	defer f.mu.RUnlock()
+
+	// helper function for determining if a host is offline. A host is
+	// considered offline if it is in allHosts but not activeHosts.
+	allHosts, activeHosts := hdb.AllHosts(), hdb.ActiveHosts()
+	isOffline := func(addr modules.NetAddress) bool {
+		for _, host := range allHosts {
+			if host.IPAddress == addr {
+				for _, host := range activeHosts {
+					if host.IPAddress == addr {
+						// host in allHosts and activeHosts
+						return false
+					}
+				}
+				// host in allHosts but not activeHosts
+				return true
+			}
+		}
+		// host not in allHosts
+		return false
+	}
```

Comment:
this for loop is potentially large enough to affect performance. AllHosts can be as large as 500 (and I'm thinking about increasing that limit), and ActiveHosts can be as large as 200 (thinking about raising it to 300). 200*500 = 10,000.

So, probably okay, but starting to be on the limit of what is acceptable for nested loops. Once we are no longer checking both all hosts and active hosts, the logic gets a lot cleaner. It's probably fine for now